### PR TITLE
Upcoming Tag for Episodes

### PR DESCRIPTION
**Pull request by KingLucius**
Show "UPCOMING" Tag for non-released Episodes.

### CODE REVIEW

File: app/src/main/res/layout/result_episode.xml [hunk 1]
```diff
@@ -90,14 +90,25 @@
...
-        android:layout_width="@dimen/download_size"
-        android:layout_height="@dimen/download_size"
-        android:layout_gravity="center_vertical|end"
-        android:layout_marginStart="-50dp"
-        android:background="?selectableItemBackgroundBorderless"
-        android:padding="10dp" />
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/episode_upcoming"
+            style="@style/SmallBlackButton"
+            android:textColor="?attr/colorPrimary"
```
[the remark this answers is not in the document]
Comment (KingLucius) [reply app/src/main/res/layout/result_episode.xml:97]: What's your suggestion is the standard?
Also, it's primary color which is not fixed. It's up to the theme color

[the remark this answers is not in the document]
Comment (KingLucius) [reply app/src/main/res/layout/result_episode.xml]: You are right, looks much better 👍

**Comment by fire-light42** [comment]
Will be merged after testing. But before that, did you ask yourself if it would look better if it said smth like `1d 15h 6m` instead of `Upcoming` as that gives more info. This is not a request to change the code, but did you think about that?

**Comment by KingLucius** [comment]
> Will be merged after testing. But before that, did you ask yourself if it would look better if it said smth like `1d 15h 6m` instead of `Upcoming` as that gives more info. This is not a request to change the code, but did you think about that?

mmmm, i think it's a nice idea
I thought about showing the air date for all episodes but i am trying redesign the card a bit to fit the airing date
And for upcoming, yes i think it will be more useful

**Comment by KingLucius** [comment]
Please hold merging i am working on another proposal

**Comment by KingLucius** [comment]
I think this can be close, I am still thinking about the suitable design for result_episode_small

**Pull request closed without merging**